cmux.go

```diff
@@ -115,7 +115,8 @@ type cMux struct {
 
 func matchersToMatchWriters(matchers []Matcher) []MatchWriter {
```

Comment:
very nice catch. we do not have a test for multiple matchers on the same listener. can you please add one? other than that LGTM.